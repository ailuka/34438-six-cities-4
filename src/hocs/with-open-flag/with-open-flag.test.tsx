import * as React from "react";
import * as renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withOpenFlag from "./with-open-flag";

const MockComponent = (props) => {
  const {isOpen} = props;
  const classType = isOpen ? `class-open` : `class-close`;

  return (
    <ul className={classType} />
  );
};

MockComponent.propTypes = {
  isOpen: PropTypes.bool.isRequired,
};

const MockComponentWrapped = withOpenFlag(MockComponent);

describe(`withOpenFlag`, () => {
  describe(`renders correctly`, () => {
    it(`when isOpen is false`, () => {
      const tree = renderer.create(
          <MockComponentWrapped/>).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});

