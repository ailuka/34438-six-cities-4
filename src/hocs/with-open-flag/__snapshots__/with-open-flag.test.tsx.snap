// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withOpenFlag renders correctly when isOpen is false 1`] = `
<ul
  className="class-close"
/>
`;
